### PR TITLE
ODRE import postprocessor fixes

### PR DESCRIPTION
**Pull request by flo-dup**
**Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**Does this PR already have an issue describing the problem?**
No

**What kind of change does this PR introduce?**
Cleaning post processor code

**What is the current behavior?**
- 40s on my laptop to import all the lines & substations positions
- 3 graph traversals for each line
- coordinates duplicated if same line reference twice with code_ligne_x and code_ligne_y
- code duplication
- substations read twice

**What is the new behavior (if this is a feature change)?**
- 5s on my laptop to import all the lines & substations positions
- graph not created if only one entry
- one single graph traversal for each problematic line 
- coordinates not duplicated if same line reference twice with code_ligne_x and code_ligne_y
- code factorization
- substations read once

**Does this PR introduce a breaking change or deprecate an API?**
- [ ] Yes
- [x] No

### CODE REVIEW

**Review by olperr1 — CHANGES_REQUESTED** [review]

File: iidm/iidm-geodata/src/main/java/com/powsybl/iidm/geodata/odre/GeographicDataParser.java [hunk 1]
```diff
@@ -266,18 +269,52 @@ private static Pair<String, String> findStartAndEndSubstationsOfLine(String line
         return Pair.of(sub1pil1 < sub2pil1 ? substation1 : substation2, sub1pil1 < sub2pil1 ? substation2 : substation1);
     }
 
-    private static List<List<Coordinate>> fillMultipleConnectedSetsCoordinatesList(List<Set<Coordinate>> connectedSets,
+    private static List<List<Coordinate>> fillMultipleConnectedSetsCoordinatesList(List<ConnectedSet> connectedSets,
```
[the remark this answers is not in the document]
Comment (flo-dup) [reply iidm/iidm-geodata/src/main/java/com/powsybl/iidm/geodata/odre/GeographicDataParser.java]: removed

File: iidm/iidm-geodata/src/main/java/com/powsybl/iidm/geodata/odre/GeographicDataParser.java [hunk 2]
```diff
@@ -167,67 +193,44 @@ private static void parseLine(Map<String, Graph<Coordinate, Object>> graphByLine
...
 
+        List<Coordinate> coordinatesComponent1 = coordinatesComponents.get(0);
+        List<Coordinate> coordinatesComponent2 = coordinatesComponents.get(1);
+
         double l1 = getBranchLength(coordinatesComponent1);
         double l2 = getBranchLength(coordinatesComponent2);
 
         if (100 * l2 / l1 < THRESHOLD) {
-            return coordinatesComponent1;
+            return l1 > l2 ? coordinatesComponent1 : coordinatesComponent2;
```
[the remark this answers is not in the document]
Comment (flo-dup) [reply iidm/iidm-geodata/src/main/java/com/powsybl/iidm/geodata/odre/GeographicDataParser.java:244]: Indeed, that was quite a mistake! fixed

**Review by olperr1 — CHANGES_REQUESTED** [review]
Could you also add a few comments in `GeographicDataParser` to describe briefly what the principal methods are doing? It will improve its readability.

File: iidm/iidm-geodata/src/main/java/com/powsybl/iidm/geodata/utils/LineCoordinatesOrdering.java [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+        } else {
+            if (s2.isPresent()) {
+                return orderCoordinates(s2.get(), lastCoordinate, firstCoordinate, lineCoordinates);
+            } else {
+                return lineCoordinates; // no information on substations positions -> considering it's in the right order
+            }
+        }
+    }
+
+    private static List<Coordinate> orderCoordinates(Coordinate s1, Coordinate s2, Coordinate firstCoordinate, Coordinate lastCoordinate,
```
[the remark this answers is not in the document]
Comment (flo-dup) [reply iidm/iidm-geodata/src/main/java/com/powsybl/iidm/geodata/utils/LineCoordinatesOrdering.java:54]: done in the ternary conditional operators, but not in the if (we don't want a warn for the equality values, which are anyway very unlikely)

File: iidm/iidm-geodata/src/main/java/com/powsybl/iidm/geodata/utils/LineCoordinatesOrdering.java [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+        Optional<Coordinate> s1 = getCoordinate(line, TwoSides.ONE);
+        Optional<Coordinate> s2 = getCoordinate(line, TwoSides.TWO);
+        Coordinate firstCoordinate = lineCoordinates.get(0);
+        Coordinate lastCoordinate = lineCoordinates.get(lineCoordinates.size() - 1);
+
+        if (s1.isPresent()) {
+            if (s2.isPresent()) {
+                return orderCoordinates(s1.get(), s2.get(), firstCoordinate, lastCoordinate, lineCoordinates, line.getId());
+            } else {
+                return orderCoordinates(s1.get(), lastCoordinate, firstCoordinate, lineCoordinates);
```
[the remark this answers is not in the document]
Comment (flo-dup) [reply iidm/iidm-geodata/src/main/java/com/powsybl/iidm/geodata/utils/LineCoordinatesOrdering.java:43]: Indeed!

**Review by olperr1 — APPROVED** [review]

**Pull request merged**